src/business_logic/transaction/fee.rs

```diff
@@ -0,0 +1,133 @@
+use super::error::TransactionError;
+use crate::{
+    business_logic::{
+        execution::{
+            execution_entry_point::ExecutionEntryPoint,
+            gas_usage,
+            objects::{CallInfo, TransactionExecutionContext},
+        },
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::cached_state::CachedState,
+    },
+    definitions::general_config::{self, StarknetGeneralConfig},
+    services::api::contract_class::EntryPointType,
+};
+use felt::Felt;
+use num_traits::{Num, ToPrimitive};
+use std::cmp;
+use std::collections::{HashMap, HashSet};
+
+// ----------------------------------------------------------------------------------------
+/// Transfers the amount actual_fee from the caller account to the sequencer.
+/// Returns the resulting CallInfo of the transfer call.
+// ----------------------------------------------------------------------------------------
+
+pub(crate) fn execute_fee_transfer(
+    state: &mut CachedState<InMemoryStateReader>,
+    general_config: StarknetGeneralConfig,
+    tx_context: TransactionExecutionContext,
+    actual_fee: u64,
+) -> Result<CallInfo, TransactionError> {
+    if actual_fee as u64 > tx_context.max_fee {
+        return Err(TransactionError::FeeError(
+            "Actual fee exceeded max fee.".to_string(),
+        ));
```

Comment:
Is the error string needed? Can't it be part of the error? Maybe even with the actual and max values.

As in the following:
```rs
enum TransactionError {
    // ...
    #[error("Actual fee exceeded max fee.")]
    FeeError,
    // ...
}
```

(there are more cases like this but I'm not commenting on every single one)

Reply:
i did it because i considered to be unnecessary to do an error for every single mistake when it is all due to the same at the end.